lib/application.js

```diff
@@ -296,7 +296,9 @@ app.enableAuth = function() {
     var req = ctx.req;
     var Model = method.ctor;
     var modelInstance = ctx.instance;
-    var modelId = modelInstance && modelInstance.id || req.param('id');
+    var modelId = (modelInstance && modelInstance.id) ||
+      (req.params && req.params.id) ||
+      (req.body && req.body.id);
```

Comment:
Note that `req.param` checked `req.body` before `req.params` as can be seen [here](https://github.com/strongloop/express/blob/4db51fbeba0cd543949e243672c0e6c3c3c70639/lib/request.js#L213-L215).

Please add a unit-test to prevent such regressions from resurfacing in the future.